plugins/chat/app/services/chat_publisher.rb

```diff
@@ -153,23 +153,19 @@ def self.publish_inaccessible_mentions(
     user_id,
     chat_message,
     cannot_chat_users,
-    without_membership
+    without_membership,
+    too_many_members,
+    mentions_disabled
   )
     MessageBus.publish(
       "/chat/#{chat_message.chat_channel_id}",
       {
         type: :mention_warning,
         chat_message_id: chat_message.id,
-        cannot_see:
-          ActiveModel::ArraySerializer.new(
-            cannot_chat_users,
-            each_serializer: BasicUserSerializer,
-          ).as_json,
-        without_membership:
-          ActiveModel::ArraySerializer.new(
-            without_membership,
-            each_serializer: BasicUserSerializer,
-          ).as_json,
+        cannot_see: cannot_chat_users.map { |u| { username: u.username, id: u.id } }.as_json,
+        without_membership: without_membership.map { |u| { username: u.username, id: u.id } }.as_json,
+        too_many_members: too_many_members.map(&:name).as_json,
```

Comment:
Im a little bit surprised by the name and behavior here: `too_many_members` I would have expected a boolean, or even just sending it if there are too many members. But it seems like it's more than this, maybe we should reflect it in the name?

Reply:
Maybe make it explicit that this is groups related? I send the name of the groups exceeding the limit because I want the error to be similar to the ones we already have. Also, one boolean is not enough since there might be multiple group mentions in the message.

Reply:
I will make another review today or tomorrow where I will try it locally, I should have better suggestions for name at this point